packages/mds-api-server/index.ts

```diff
@@ -9,7 +9,18 @@ import { Params, ParamsDictionary } from 'express-serve-static-core'
 
 export type ApiRequest<P extends Params = ParamsDictionary> = express.Request<P>
 
-export type ApiQuery<Q extends string> = { query: Partial<{ [P in Q]: string | string[] }> }
+export type ApiQuery<Q1 extends string, Q2 extends string[] = never> = {
+  query: Partial<
+    {
+      [P in Q1]: string
```

Comment:
This syntax is new to me, can you give a little explanation?

Reply:
Basically, there is an optional `string` property on the object for every name in Q1 and an optional string or array of strings property on the object for every name in Q2. Any property that isn't specified can not be used. Forces type safe access.